client/src/partials/records/sales_records/sales_records.js

```diff
@@ -142,7 +142,12 @@ angular.module('bhima.controllers')
 
     function totalCost() {
       return $scope.model.sale.data.reduce(function (a, b) {
```

Comment:
True, but I am more worried about code clarity at this point than speed.  Both the current loop and my suggested loop are `O(n)`, which are theoretically the same speed.